frontend2/src/pages/tables/employee-table.tsx

```diff
@@ -0,0 +1,89 @@
+import { EmployeeTableColumns } from '../../components/tableColumns/EmployeeTableColumns';
+
+export const EmpTable: React.FC = observer((): JSX.Element => {
+    const rootStore = useRootStore();
+
+    const data = [...rootStore.shiftStore.employees];
+
+    const cols = EmployeeTableColumns(rootStore.employeeStore, rootStore.companyStore);
+
+    const columns = React.useMemo(() => cols.map((col) => ({ ...col, accessor: col.key, Header: col.title })), []);
+
+    const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = useTable({
+        columns,
+        data: data,
+        initialState: rootStore.shiftStore.employees,
+    });
+
+    return (
+        <table {...getTableProps()} style={{ border: 'solid 1px blue' }}>
+            <thead>
+                {headerGroups.map((headerGroup, index) => (
+                    <tr {...headerGroup.getHeaderGroupProps()} key={`employee-table-thead-tr-${index}`}>
+                        {headerGroup.headers.map((column) => (
+                            <th
+                                key={`employee-table-thead-th-${index}`}
+                                {...column.getHeaderProps()}
+                                style={{
+                                    borderBottom: 'solid 3px red',
+                                    background: 'aliceblue',
+                                    color: 'black',
+                                    fontWeight: 'bold',
+                                }}
+                            >
+                                {column.render('Header')}
+                            </th>
+                        ))}
+                    </tr>
+                ))}
+            </thead>
+
+            <Droppable droppableId="employee-table">
+                {(provided, snapshot) => (
+                    <tbody {...getTableBodyProps()} ref={provided.innerRef} {...provided.droppableProps}>
+                        {rows.map((row, index) => {
+                            prepareRow(row);
+                            return (
+                                <Draggable draggableId={row.id.toString()} key={row.id} index={row.index}>
+                                    {(provided, snapshot) => {
+                                        return (
+                                            <tr
+                                                key={`employee-table-tbody-tr-${index}`}
+                                                {...row.getRowProps()}
+                                                {...provided.draggableProps}
+                                                {...provided.dragHandleProps}
+                                                ref={provided.innerRef}
+                                            >
+                                                {row.cells.map((cell, index) => {
+                                                    return (
+                                                        <td

```

Comment:
I would suggest extracting some of these subcomponents to separate functions (=functional components). This level of indentation is quite unreadable :)

Reply:
Yes, I definitelly plan on doing that. It is still in testing phase in the moment... :D